src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -73,6 +73,20 @@ trait WdlStandardLibraryFunctions extends WdlFunctions[WdlValue] {
     } yield WdlArray(WdlArrayType(WdlFileType), wdlFiles)
   }
 
+  def select_first(params: Seq[Try[WdlValue]]): Try[WdlValue] = extractSingleArgument(params) flatMap {
+    case WdlArray(WdlArrayType(WdlOptionalType(_), _), arrayValue) =>
+      (arrayValue collect { case WdlOptionalValue(_, Some(wdlValue)) => wdlValue }).headOption
```

Comment:
are the outermost parens here only necessary because you're avoiding `arrayValue.collect(.....`? If so I'd just convert to the latter as the rest of the line is using that structure